modules/effects/spec/effects_error_handler.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { ErrorHandler, Provider } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { Action, Store } from '@ngrx/store';
+import { Observable, of } from 'rxjs';
+import { catchError } from 'rxjs/operators';
+import { createEffect, EFFECTS_ERROR_HANDLER, EffectsModule } from '..';
+
+describe('NgRx Effects Error Handler spec', () => {
+  let subscriptionCount: number;
+  let globalErrorHandler: jasmine.Spy;
+  let storeNext: jasmine.Spy;
+
+  function makeEffectTestBed(...providers: Provider[]) {
```

Comment:
So I tried to do this refactor and ended up having to create a bunch of duplicate code, which is a step backwards in my opinion. This test is somewhat unusual compared to other specs in that all the behavior happens on startup (the only difference in the two tests is the providers, which can only be set when initialising the whole test bed.
To use beforeEach, I would have to completely re-declare the `TestBed.configureTestingModule` in separate beforeEach blocks.

I think it is safer to have the spec focussed on what is different between each test rather than following the pattern used elsewhere.

Reply:
I missed the `providers` argument. 
👍